utils/gas-utils/gas_utils.go

```diff
@@ -23,13 +26,24 @@ const (
 // depends on the required limit for the message execution, the number of validator signatures
 // included in the aggregate signature, the static gas cost defined by the precompile, and an
 // extra buffer amount defined here to ensure the call doesn't run out of gas.
-func CalculateReceiveMessageGasLimit(numSigners int, executionRequiredGasLimit *big.Int) (uint64, error) {
+func CalculateReceiveMessageGasLimit(
+	numSigners int,
+	executionRequiredGasLimit *big.Int,
+	numWarpMessageBytes int,
+	teleporterMessage teleportermessenger.TeleporterMessage) (uint64, error) {
 	if !executionRequiredGasLimit.IsUint64() {
 		return 0, errors.New("required gas limit too high")
 	}
 
 	gasAmounts := []uint64{
 		executionRequiredGasLimit.Uint64(),
+		// The variable gas on message bytes is accounted for both when used in predicate verification
+		// and also when used in `getVerifiedWarpMessage`
+		uint64(numWarpMessageBytes) * warp.GasCostPerWarpMessageBytes * 2,
+		// Take into the variable gas cost for decoding the Teleporter message
+		// and marking the receipts as received.
+		uint64(len(teleporterMessage.Message)) * DecodeGasPerByte,
```

Comment:
There are 3 variable length parts of a teleporter message:
- The payload
- The receipts array (has a maximum of 5 * size of a single receipt)
- The `allowedRelayerAddresses`


Technically each of those could increase the gas code of the `abi.decode` call (although we're separately accounting for the number of receipts already), so should we account for them all here?